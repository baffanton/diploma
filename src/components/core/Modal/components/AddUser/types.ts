export interface IAddUserModal {
    onClose: () => void;
    option: IAddUserModalOptions;
}

export interface IAddUserModalOptions {
    onAddUserHandler: (id: string, data: IAddUserDataModel) => void;
}

export interface IAddUserDataModel {
    firstname: string;
    lastname: string;
    surname: string;
    workPlace: string;
    position: string;
    phone: string;
}
