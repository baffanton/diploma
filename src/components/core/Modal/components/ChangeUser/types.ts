import { ChangeUserDataTypes } from 'enums/changeUserDataTypes';

export interface IChangeUserModal {
    option: IChangeUserModalOptions;
    onClose: () => void;
}

export interface IChangeUserModalOptions {
    type: ChangeUserDataTypes;
    user?: any;
    onSubmitHandler: (id: string, data: IChangeUserDataModel) => void;
}

export interface IChangeUserDataModel {
    firstname: string;
    lastname: string;
    surname: string;
    workPlace: string;
    position: string;
    phone: string;
}
