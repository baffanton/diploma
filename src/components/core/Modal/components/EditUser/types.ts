export interface IEditUserModel {
    option: IEditUserModalOptions;
    onClose: () => void;
}

export interface IEditUserModalOptions {
    user: any;
    onEditUserHandler: (id: string, data: IEditUserDataModel) => void;
}

export interface IEditUserDataModel {
    id: string;
    firstname: string;
    lastname: string;
    surname: string;
    workPlace: string;
    position: string;
    phone: string;
}
