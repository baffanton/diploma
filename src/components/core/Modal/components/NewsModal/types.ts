export interface INewsModal {
    onClose: () => void;
    option: INewsModalOptions;
}

export interface INewsModalOptions {
    title: string;
    description: string;
    source: {
        vk: string;
        tg: string;
    };
    picture: string;
}
