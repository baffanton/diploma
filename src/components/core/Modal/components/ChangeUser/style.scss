@import 'global.scss';

.change-user-modal {
    width: 1100px;
    background-color: $white;
    border-radius: 20px;
}

.change-user-modal__form {
    box-sizing: border-box;
}

.change-user-modal__header {
    display: flex;
    justify-content: center;
    padding: 16px;
    background-color: $darkBlue;
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
    z-index: 1;
    position: relative;
}

.change-user-modal__title {
    color: $white;
}

.change-user-modal__close {
    position: absolute;
    right: 16px;
    color: $white;
}

.change-user-modal__body {
    padding: 0 32px;
    display: flex;
    flex-direction: column;
}

.change-user-modal__panel {
    display: flex;
    width: 100%;
}

.change-user-modal__personal-data,
.change-user-modal__work-info {
    width: 50%;
    padding: 0 10px;
}

.change-user-modal__block-title {
    padding: 16px 16px 8px 16px;
    text-align: center;
    color: $darkBlue;
}

.change-user-modal__lastname,
.change-user-modal__name,
.change-user-modal__username,
.change-user-modal__work-place,
.change-user-modal__position,
.change-user-modal__phone {
    margin-bottom: 8px;
}

.change-user-modal__submit-button-container {
    margin: 20px 10px 30px 10px;
}

.change-user-modal__submit-button {
    width: 100%;
}
