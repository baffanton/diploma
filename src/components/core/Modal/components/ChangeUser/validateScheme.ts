import * as yup from 'yup';

export const schema = yup.object({
    firstname: yup.string().required('Поле не может быть пустым*'),
    lastname: yup.string().required('Поле не может быть пустым*'),
    surname: yup.string().required('Поле не может быть пустым*'),
    workPlace: yup.string().required('Поле не может быть пустым*'),
    position: yup.string().required('Поле не может быть пустым*'),
    phone: yup
        .string()
        .required('Поле не может быть пустым*')
        .min(11, 'Не менее 11 символов')
        .max(11, 'Не более 11 символов'),
});
