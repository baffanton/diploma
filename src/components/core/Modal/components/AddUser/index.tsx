import React from 'react';
import { SubmitHandler, useForm } from 'react-hook-form';

import { Button } from 'components/core/Button';
import { Icon } from 'components/core/Icon';
import { Layout } from 'components/core/Layout';
import { Text } from 'components/core/Text';
import { TextBox } from 'components/core/TextBox';

import { IAddUserDataModel, IAddUserModal } from './types';
import { schema } from './validateScheme';
import { faXmark } from '@fortawesome/free-solid-svg-icons';
import { yupResolver } from '@hookform/resolvers/yup';
import { ColorThemeType } from 'enums/colorThemeTypes';
import { LabelPositionEnum } from 'enums/labelPositionTypes';
import { SizeEnum } from 'enums/sizeTypes';
import { WeightEnum } from 'enums/weightTypes';

import './style.scss';

const AddUserModal: React.FC<IAddUserModal> = ({ onClose, option }) => {
    const {
        register,
        handleSubmit,
        formState: { errors },
    } = useForm<IAddUserDataModel>({
        mode: 'onSubmit',
        resolver: yupResolver(schema),
    });

    const { onAddUserHandler } = option;

    const onSubmit: SubmitHandler<IAddUserDataModel> = (formData) => {
        onAddUserHandler(formData);
    };

    return (
        <Layout className="add-user-modal">
            <Layout className="add-user-modal__header">
                <Text className="add-user-modal__title" fontSize={SizeEnum.large}>
                    Новый пользователь
                </Text>
                <Icon
                    className="add-user-modal__close"
                    onClick={() => onClose()}
                    fontAwesomeIcon={faXmark}
                    heightType={SizeEnum.short}
                />
            </Layout>
            <Layout className="add-user-modal__body">
                <form className="add-user-modal__form" onSubmit={handleSubmit(onSubmit)}>
                    <Layout className="add-user-modal__personal-data">
                        <Text className="add-user-modal__block-title" fontWeight={WeightEnum.bold}>
                            Личные данные
                        </Text>
                        <TextBox
                            id="lastname"
                            label="Фамилия"
                            labelPosition={LabelPositionEnum.top}
                            placeholder="Введите фамилию"
                            register={register('lastname')}
                            error={errors.lastname}
                            colorTheme={ColorThemeType.primary}
                            heightType={SizeEnum.short}
                            classNameContainer="add-user-modal__lastname"
                        />
                        <TextBox
                            id="name"
                            label="Имя"
                            labelPosition={LabelPositionEnum.top}
                            placeholder="Введите имя"
                            register={register('firstname')}
                            error={errors.firstname}
                            colorTheme={ColorThemeType.primary}
                            heightType={SizeEnum.short}
                            classNameContainer="add-user-modal__name"
                        />
                        <TextBox
                            id="surname"
                            label="Отчество"
                            labelPosition={LabelPositionEnum.top}
                            placeholder="Введите отчество"
                            register={register('surname')}
                            error={errors.surname}
                            colorTheme={ColorThemeType.primary}
                            heightType={SizeEnum.short}
                            classNameContainer="add-user-modal__surname"
                        />
                    </Layout>
                    <Layout className="add-user-modal__login-details">
                        <Text className="add-user-modal__block-title" fontWeight={WeightEnum.bold}>
                            Данные для входа
                        </Text>
                        <TextBox
                            id="username"
                            label="Имя пользователя"
                            labelPosition={LabelPositionEnum.top}
                            placeholder="Введите имя пользователя"
                            register={register('username')}
                            error={errors.username}
                            colorTheme={ColorThemeType.primary}
                            heightType={SizeEnum.short}
                            classNameContainer="add-user-modal__username"
                        />
                        <TextBox
                            id="password"
                            label="Пароль"
                            labelPosition={LabelPositionEnum.top}
                            placeholder="Введите пароль"
                            register={register('password')}
                            error={errors.password}
                            colorTheme={ColorThemeType.primary}
                            heightType={SizeEnum.short}
                            classNameContainer="add-user-modal__password"
                        />
                    </Layout>
                    <Layout className="add-user-modal__submit-button-container">
                        <Button
                            className="add-user-modal__submit-button"
                            colorTheme={ColorThemeType.primary}
                            heightType={SizeEnum.medium}
                            onClick={() => null}
                        >
                            Добавить пользователя
                        </Button>
                    </Layout>
                </form>
            </Layout>
        </Layout>
    );
};

export { AddUserModal };
