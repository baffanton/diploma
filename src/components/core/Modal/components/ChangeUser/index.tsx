import React from 'react';
import { SubmitHandler, useForm } from 'react-hook-form';
import { Button } from 'components/core/Button';
import { Icon } from 'components/core/Icon';
import { Layout } from 'components/core/Layout';
import { Text } from 'components/core/Text';
import { TextBox } from 'components/core/TextBox';
import { IChangeUserDataModel, IChangeUserModal } from './types';
import { schema } from './validateScheme';
import { faXmark } from '@fortawesome/free-solid-svg-icons';
import { yupResolver } from '@hookform/resolvers/yup';
import { ChangeUserDataTypes } from 'enums/changeUserDataTypes';
import { ColorThemeType } from 'enums/colorThemeTypes';
import { LabelPositionEnum } from 'enums/labelPositionTypes';
import { SizeEnum } from 'enums/sizeTypes';
import { WeightEnum } from 'enums/weightTypes';
import { v4 as uuid } from 'uuid';
import './style.scss';

const ChangeUserModal: React.FC<IChangeUserModal> = ({ option, onClose }) => {
    const { type, user, onSubmitHandler } = option;
    const isEdit = type === ChangeUserDataTypes.edit;

    const {
        register,
        handleSubmit,
        formState: { errors },
    } = useForm<IChangeUserDataModel>({
        mode: 'onSubmit',
        defaultValues: {
            firstname: isEdit ? user.firstname : '',
            lastname: isEdit ? user.lastname : '',
            surname: isEdit ? user.surname : '',
            workPlace: isEdit ? user.workPlace : '',
            position: isEdit ? user.position : '',
            phone: isEdit ? user.phone : '',
        },
        resolver: yupResolver(schema),
    });

    const onSubmit: SubmitHandler<IChangeUserDataModel> = (formData) => {
        if (isEdit) {
            onSubmitHandler(user.id, formData);
        }

        const id = uuid();
        onSubmitHandler(id, formData);
    };

    return (
        <Layout className="change-user-modal">
            <Layout className="change-user-modal__header">
                <Text className="change-user-modal__title" fontSize={SizeEnum.large}>
                    {isEdit ? 'Изменение данных пользователя' : 'Новый пользователь'}
                </Text>
                <Icon
                    className="change-user-modal__close"
                    onClick={() => onClose()}
                    fontAwesomeIcon={faXmark}
                    heightType={SizeEnum.short}
                />
            </Layout>
            <Layout className="change-user-modal__body">
                <form className="change-user-modal__form" onSubmit={handleSubmit(onSubmit)}>
                    <Layout className="change-user-modal__panel">
                        <Layout className="change-user-modal__personal-data">
                            <Text className="change-user-modal__block-title" fontWeight={WeightEnum.bold}>
                                Личные данные
                            </Text>
                            <TextBox
                                id="lastname"
                                label="Фамилия"
                                labelPosition={LabelPositionEnum.top}
                                placeholder="Введите фамилию"
                                register={register('lastname')}
                                error={errors.lastname}
                                colorTheme={ColorThemeType.primary}
                                heightType={SizeEnum.medium}
                                classNameContainer="change-user-modal__lastname"
                            />
                            <TextBox
                                id="name"
                                label="Имя"
                                labelPosition={LabelPositionEnum.top}
                                placeholder="Введите имя"
                                register={register('firstname')}
                                error={errors.firstname}
                                colorTheme={ColorThemeType.primary}
                                heightType={SizeEnum.medium}
                                classNameContainer="change-user-modal__name"
                            />
                            <TextBox
                                id="surname"
                                label="Отчество"
                                labelPosition={LabelPositionEnum.top}
                                placeholder="Введите отчество"
                                register={register('surname')}
                                error={errors.surname}
                                colorTheme={ColorThemeType.primary}
                                heightType={SizeEnum.medium}
                                classNameContainer="change-user-modal__surname"
                            />
                        </Layout>
                        <Layout className="change-user-modal__work-info">
                            <Text className="change-user-modal__block-title" fontWeight={WeightEnum.bold}>
                                Данные о работе
                            </Text>
                            <TextBox
                                id="workPlace"
                                label="Место работы"
                                labelPosition={LabelPositionEnum.top}
                                placeholder="Введите место работы"
                                register={register('workPlace')}
                                error={errors.workPlace}
                                colorTheme={ColorThemeType.primary}
                                heightType={SizeEnum.medium}
                                classNameContainer="change-user-modal__work-place"
                            />
                            <TextBox
                                id="position"
                                label="Должность"
                                labelPosition={LabelPositionEnum.top}
                                placeholder="Введите должность"
                                register={register('position')}
                                error={errors.position}
                                colorTheme={ColorThemeType.primary}
                                heightType={SizeEnum.medium}
                                classNameContainer="change-user-modal__position"
                            />
                            <TextBox
                                id="phone"
                                label="Номер телефона"
                                labelPosition={LabelPositionEnum.top}
                                placeholder="Введите номер телефона"
                                register={register('phone')}
                                error={errors.phone}
                                colorTheme={ColorThemeType.primary}
                                heightType={SizeEnum.medium}
                                classNameContainer="change-user-modal__phone"
                            />
                        </Layout>
                    </Layout>
                    <Layout className="change-user-modal__submit-button-container">
                        <Button
                            className="change-user-modal__submit-button"
                            colorTheme={ColorThemeType.primary}
                            heightType={SizeEnum.medium}
                            onClick={() => null}
                        >
                            {isEdit ? 'Изменить данные' : 'Добавить пользователя'}
                        </Button>
                    </Layout>
                </form>
            </Layout>
        </Layout>
    );
};

export { ChangeUserModal };
