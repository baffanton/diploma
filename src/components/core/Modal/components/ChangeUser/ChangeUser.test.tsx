import React from 'react';
import { ChangeUserModal } from 'components/core/Modal/components/ChangeUser';
import { IChangeUserDataModel, IChangeUserModalOptions } from './types';
import { act, fireEvent, render } from '@testing-library/react';
import { ChangeUserDataTypes } from 'enums/changeUserDataTypes';

describe('ChangeUserModal tests', () => {
    it('calls the onSubmit function', async () => {
        const mockSubmitHandler = jest.fn();

        const mockOptions: IChangeUserModalOptions = {
            type: ChangeUserDataTypes.add,
            onSubmitHandler: (id: string, data: IChangeUserDataModel) => mockSubmitHandler(id, data),
        };

        const { getByLabelText, getByRole } = render(<ChangeUserModal onClose={() => null} option={mockOptions} />);

        await act(async () => {
            fireEvent.change(getByLabelText('Фамилия'), { target: { value: 'Баяндин' } });
            fireEvent.change(getByLabelText('Имя'), { target: { value: 'Антон' } });
            fireEvent.change(getByLabelText('Отчество'), { target: { value: 'Викторович' } });
            fireEvent.change(getByLabelText('Место работы'), { target: { value: 'АБК-2' } });
            fireEvent.change(getByLabelText('Должность'), { target: { value: 'Директор' } });
            fireEvent.change(getByLabelText('Номер телефона'), { target: { value: '89824467190' } });
        });

        await act(async () => {
            fireEvent.click(getByRole('button'));
        });

        expect(mockSubmitHandler).toHaveBeenCalled();
    });
    it('with invalid fields', async () => {
        const mockSubmitHandler = jest.fn();

        const mockOptions: IChangeUserModalOptions = {
            type: ChangeUserDataTypes.add,
            onSubmitHandler: (id: string, data: IChangeUserDataModel) => mockSubmitHandler(id, data),
        };

        const { container, getByLabelText, getByRole } = render(
            <ChangeUserModal onClose={() => null} option={mockOptions} />,
        );

        await act(async () => {
            fireEvent.change(getByLabelText('Фамилия'), { target: { value: 'Баяндин' } });
            fireEvent.change(getByLabelText('Имя'), { target: { value: 'Антон' } });
            fireEvent.change(getByLabelText('Отчество'), { target: { value: 'Викторович' } });
            fireEvent.change(getByLabelText('Место работы'), { target: { value: 'АБК-2' } });
        });

        await act(async () => {
            fireEvent.click(getByRole('button'));
        });

        expect(container.getElementsByClassName('text-box__error').length).toBe(2);
    });
    it('Modal snapshot', () => {
        const mockOptions: IChangeUserModalOptions = {
            type: ChangeUserDataTypes.add,
            onSubmitHandler: () => null,
        };

        const modal = render(<ChangeUserModal onClose={() => null} option={mockOptions} />);

        expect(modal).toMatchSnapshot();
    });
});
