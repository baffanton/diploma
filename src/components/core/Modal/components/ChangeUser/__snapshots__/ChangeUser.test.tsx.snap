// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeUserModal tests Modal snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout change-user-modal"
      >
        <div
          class="layout change-user-modal__header"
        >
          <p
            class="text text_size_l text_style_default text_weight_default change-user-modal__title"
          >
            Новый пользователь
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-xmark icon icon_pointer icon_heightType_s change-user-modal__close"
            data-icon="xmark"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="layout change-user-modal__body"
        >
          <form
            class="change-user-modal__form"
          >
            <div
              class="layout change-user-modal__panel"
            >
              <div
                class="layout change-user-modal__personal-data"
              >
                <p
                  class="text text_size_m text_style_default text_weight_bold change-user-modal__block-title"
                >
                  Личные данные
                </p>
                <div
                  class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__lastname"
                >
                  <div
                    class="layout text-box__label-container text-box__label-container_position_top"
                  >
                    <label
                      class="text-box__label text-box__label_color_primary text-box__label_height_m"
                      for="lastname"
                    >
                      Фамилия
                    </label>
                  </div>
                  <input
                    autocomplete="off"
                    class="text-box__input text-box__input_color_primary text-box__input_height_m"
                    id="lastname"
                    name="lastname"
                    placeholder="Введите фамилию"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__name"
                >
                  <div
                    class="layout text-box__label-container text-box__label-container_position_top"
                  >
                    <label
                      class="text-box__label text-box__label_color_primary text-box__label_height_m"
                      for="name"
                    >
                      Имя
                    </label>
                  </div>
                  <input
                    autocomplete="off"
                    class="text-box__input text-box__input_color_primary text-box__input_height_m"
                    id="name"
                    name="firstname"
                    placeholder="Введите имя"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__surname"
                >
                  <div
                    class="layout text-box__label-container text-box__label-container_position_top"
                  >
                    <label
                      class="text-box__label text-box__label_color_primary text-box__label_height_m"
                      for="surname"
                    >
                      Отчество
                    </label>
                  </div>
                  <input
                    autocomplete="off"
                    class="text-box__input text-box__input_color_primary text-box__input_height_m"
                    id="surname"
                    name="surname"
                    placeholder="Введите отчество"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="layout change-user-modal__work-info"
              >
                <p
                  class="text text_size_m text_style_default text_weight_bold change-user-modal__block-title"
                >
                  Данные о работе
                </p>
                <div
                  class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__work-place"
                >
                  <div
                    class="layout text-box__label-container text-box__label-container_position_top"
                  >
                    <label
                      class="text-box__label text-box__label_color_primary text-box__label_height_m"
                      for="workPlace"
                    >
                      Место работы
                    </label>
                  </div>
                  <input
                    autocomplete="off"
                    class="text-box__input text-box__input_color_primary text-box__input_height_m"
                    id="workPlace"
                    name="workPlace"
                    placeholder="Введите место работы"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__position"
                >
                  <div
                    class="layout text-box__label-container text-box__label-container_position_top"
                  >
                    <label
                      class="text-box__label text-box__label_color_primary text-box__label_height_m"
                      for="position"
                    >
                      Должность
                    </label>
                  </div>
                  <input
                    autocomplete="off"
                    class="text-box__input text-box__input_color_primary text-box__input_height_m"
                    id="position"
                    name="position"
                    placeholder="Введите должность"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__phone"
                >
                  <div
                    class="layout text-box__label-container text-box__label-container_position_top"
                  >
                    <label
                      class="text-box__label text-box__label_color_primary text-box__label_height_m"
                      for="phone"
                    >
                      Номер телефона
                    </label>
                  </div>
                  <input
                    autocomplete="off"
                    class="text-box__input text-box__input_color_primary text-box__input_height_m"
                    id="phone"
                    name="phone"
                    placeholder="Введите номер телефона"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="layout change-user-modal__submit-button-container"
            >
              <button
                class="button button_height_m button_theme_primary change-user-modal__submit-button"
              >
                Добавить пользователя
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout change-user-modal"
    >
      <div
        class="layout change-user-modal__header"
      >
        <p
          class="text text_size_l text_style_default text_weight_default change-user-modal__title"
        >
          Новый пользователь
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark icon icon_pointer icon_heightType_s change-user-modal__close"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="layout change-user-modal__body"
      >
        <form
          class="change-user-modal__form"
        >
          <div
            class="layout change-user-modal__panel"
          >
            <div
              class="layout change-user-modal__personal-data"
            >
              <p
                class="text text_size_m text_style_default text_weight_bold change-user-modal__block-title"
              >
                Личные данные
              </p>
              <div
                class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__lastname"
              >
                <div
                  class="layout text-box__label-container text-box__label-container_position_top"
                >
                  <label
                    class="text-box__label text-box__label_color_primary text-box__label_height_m"
                    for="lastname"
                  >
                    Фамилия
                  </label>
                </div>
                <input
                  autocomplete="off"
                  class="text-box__input text-box__input_color_primary text-box__input_height_m"
                  id="lastname"
                  name="lastname"
                  placeholder="Введите фамилию"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__name"
              >
                <div
                  class="layout text-box__label-container text-box__label-container_position_top"
                >
                  <label
                    class="text-box__label text-box__label_color_primary text-box__label_height_m"
                    for="name"
                  >
                    Имя
                  </label>
                </div>
                <input
                  autocomplete="off"
                  class="text-box__input text-box__input_color_primary text-box__input_height_m"
                  id="name"
                  name="firstname"
                  placeholder="Введите имя"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__surname"
              >
                <div
                  class="layout text-box__label-container text-box__label-container_position_top"
                >
                  <label
                    class="text-box__label text-box__label_color_primary text-box__label_height_m"
                    for="surname"
                  >
                    Отчество
                  </label>
                </div>
                <input
                  autocomplete="off"
                  class="text-box__input text-box__input_color_primary text-box__input_height_m"
                  id="surname"
                  name="surname"
                  placeholder="Введите отчество"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="layout change-user-modal__work-info"
            >
              <p
                class="text text_size_m text_style_default text_weight_bold change-user-modal__block-title"
              >
                Данные о работе
              </p>
              <div
                class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__work-place"
              >
                <div
                  class="layout text-box__label-container text-box__label-container_position_top"
                >
                  <label
                    class="text-box__label text-box__label_color_primary text-box__label_height_m"
                    for="workPlace"
                  >
                    Место работы
                  </label>
                </div>
                <input
                  autocomplete="off"
                  class="text-box__input text-box__input_color_primary text-box__input_height_m"
                  id="workPlace"
                  name="workPlace"
                  placeholder="Введите место работы"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__position"
              >
                <div
                  class="layout text-box__label-container text-box__label-container_position_top"
                >
                  <label
                    class="text-box__label text-box__label_color_primary text-box__label_height_m"
                    for="position"
                  >
                    Должность
                  </label>
                </div>
                <input
                  autocomplete="off"
                  class="text-box__input text-box__input_color_primary text-box__input_height_m"
                  id="position"
                  name="position"
                  placeholder="Введите должность"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="layout text-box__wrapper text-box__wrapper_label_top change-user-modal__phone"
              >
                <div
                  class="layout text-box__label-container text-box__label-container_position_top"
                >
                  <label
                    class="text-box__label text-box__label_color_primary text-box__label_height_m"
                    for="phone"
                  >
                    Номер телефона
                  </label>
                </div>
                <input
                  autocomplete="off"
                  class="text-box__input text-box__input_color_primary text-box__input_height_m"
                  id="phone"
                  name="phone"
                  placeholder="Введите номер телефона"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="layout change-user-modal__submit-button-container"
          >
            <button
              class="button button_height_m button_theme_primary change-user-modal__submit-button"
            >
              Добавить пользователя
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
